torchbenchmark/models/tacotron2/__init__.py

```diff
@@ -19,130 +22,141 @@ def __init__(self, device=None, jit=False):
             # TODO - currently load_model assumes cuda
             return
 
-        self.hparams = self.create_hparams()
-        self.model = load_model(self.hparams).to(device=device)
-        self.optimizer = torch.optim.Adam(self.model.parameters(), lr=self.hparams.learning_rate,
-                                 weight_decay=self.hparams.weight_decay)
+        self.train_hparams = self.create_hparams(batch_size=train_bs)
+        self.eval_hparams = self.create_hparams(batch_size=eval_bs)
+        self.train_model = load_model(self.train_hparams).to(device=device)
+        self.eval_model = load_model(self.eval_hparams).to(device=device)
+        self.train_optimizer = torch.optim.Adam(self.train_model.parameters(),
+                                                lr=self.train_hparams.learning_rate,
+                                                weight_decay=self.train_hparams.weight_decay)
+        self.eval_optimizer = torch.optim.Adam(self.eval_model.parameters(),
+                                               lr=self.eval_hparams.learning_rate,
+                                               weight_decay=self.eval_hparams.weight_decay)
         self.criterion = Tacotron2Loss().to(device=device)
-        train_loader, valset, collate_fn = prepare_dataloaders(self.hparams)
-        self.example_input, self.target = self.model.parse_batch(list(train_loader)[0], device=self.device)
-
+        train_loader, valset, collate_fn = prepare_dataloaders(self.train_hparams)
```

Comment:
Is it possible to add an option to `__init__()` that preloads the data and reprofile the workload? Would like to see if the bad GPU utilization is caused by dataloader.

Reply:
Good idea, I think it can be added in a separate PR.